docs/common.md

```diff
@@ -0,0 +1,102 @@
+# Common
+
+## Important Key Ideas
+
+### Versioning
+
+
+
+### Data, Representation and Encoding independence
+
+Everything in the interchange format should have a clear distinction between;
+
+ * (a) the data "as a concept",
+ * (b) the data "as represented" and
+ * (c) the data "as encoded in a file format".
+
+For example,
+
+ * (a) the data being represented could be "the name of an object",
+ * (b) but it could be represented as an integer pointing to a UTF-8 string
+   table, and
+ * (c) that could be encoded as either XML or Cap'n'Proto file format.
+
+### On disk representation
+
+The interchange format should define both;
+
+ * (a) A compact binary machine readable format, **and**
+ * (b) a texted based human readable format.
+
+Tools should exist which do lossless conversion between the machine and human
+readable formats.
+
+The preferred on disk formats for the interchange format are;
+
+ * (a) Binary Machine readable format - **Cap'n'Proto**
+ * (b) Text based human readable format - **XML**
```

Comment:
It's worth noting that we don't actually have any XML support at all atm, and seem to be preferring YAML for human readable/writeable stuff like the constraint/LUT patches for xc7, so far

Reply:
Perhaps it is worth mentioning that one of the original motivations of having both a binary and text based format were to provide the need for speed/space efficiency and debuggability.  Binary has much better speed and data compaction properties whereas text enables powerful checks (such as diff).  

The most readable/debugging-useful text format would probably be something custom, however that introduces additional overhead of building custom parsers/deparsers for each language.  The ecosystem of tooling that already exists around YAML, XML and JSON is I think one of the primary motivators for their selection.